challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map {|a| a.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select {|a| a.casecmp("dog") == 0}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject {|a| a.casecmp("dog") == 0}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep(/^[cdCD].*/)
```

Comment:
`/^[cdCD]/` would also match strings starting with "c" or "d".

Also, the regex can be marked as "case-insensitive like so: `/^[cd]/i`.